client/notifications/activity-bar-entry.tsx

```diff
@@ -80,17 +85,26 @@ const PopoverContents = styled.div<PopoverContentsProps>`
 
 export function NotificationsButton() {
   const dispatch = useAppDispatch()
-  const notifications = useAppSelector(s => s.notifications.list)
-  const hasUnread = useMemo(() => notifications.some(n => n.unread), [notifications])
+  const map = useAppSelector(s => s.notifications.map)
+  const ids = useAppSelector(s => s.notifications.ids)
+  const group = ids.groupBy(id => map.get(id)?.local)
+  const localNotifications = group.get(true)
+  const serverNotifications = group.get(false)
+  const hasUnread = useMemo(() => map.some(n => !n.read), [map])
 
   const [anchor, setAnchor] = useState<EventTarget | null>(null)
   const onClick = useCallback((event: React.MouseEvent) => {
     setAnchor(event.currentTarget)
   }, [])
   const onDismiss = useCallback(() => {
     setAnchor(null)
-    dispatch(markNotificationsRead())
-  }, [dispatch])
+    if (localNotifications?.count()) {
```

Comment:
Apparently, the `groupBy` function returns a [Keyed Collection](https://immutable-js.github.io/immutable-js/docs/#/Collection/Keyed) data structure which doesn't seem to have the `size` property as far as I can tell. Not sure if using the `count()` method is correct here or do we need to transform the Collection to something else first, that has a `size` property.

Reply:
That's fine then, sorry, I thought this was returning a `Map` or similar. `size` is preferred since it's always constant time, but if it requires a data structure conversion it's not worth avoiding `count()`.